### PR TITLE
ui: Fix missing thread relation when sending reply to threaded message

### PR DESCRIPTION
**Pull request by jplatte**
https://github.com/vector-im/element-x-ios/issues/1883

### CODE REVIEW

File: crates/matrix-sdk-ui/src/timeline/event_item/content/message.rs [hunk 1]
```diff
@@ -79,13 +83,13 @@ impl Message {
...
 
-        let mut threaded = false;
+        let mut thread_root = None;
         let in_reply_to = c.relates_to.and_then(|relation| match relation {
             message::Relation::Reply { in_reply_to } => {
                 Some(InReplyToDetails::new(in_reply_to.event_id, timeline_items))
             }
             message::Relation::Thread(thread) => {
-                threaded = true;
+                thread_root = Some(thread.event_id);
```
Comment (bnjbvr) [inline crates/matrix-sdk-ui/src/timeline/event_item/content/message.rs:92]: Is this always guaranteed to be the thread root? (and not a message in the middle of the thread)

**Review by bnjbvr — APPROVED** [review]
Nice, thanks!

File: crates/matrix-sdk-ui/src/timeline/event_item/content/message.rs (hunk 1 above)
Comment (jplatte) [reply crates/matrix-sdk-ui/src/timeline/event_item/content/message.rs:92]: Yes, see https://spec.matrix.org/v1.8/client-server-api/#threading.

Comment (bnjbvr) [reply crates/matrix-sdk-ui/src/timeline/event_item/content/message.rs:92]: TIL :ok_hand:

**Pull request merged**